pkg/kudoctl/util/repo/repo_framework.go

```diff
@@ -0,0 +1,236 @@
+	}
+
+	index, err := ioutil.ReadAll(resp)
+	if err != nil {
+		return errors.Wrap(err, "reading index response")
+	}
+
+	if _, err := loadIndex(index); err != nil {
+		return errors.Wrap(err, "loading index file")
+	}
+
+	if filepath.IsAbs(path) {
+		path = filepath.Join(path, "index.yaml")
+	}
+
+	return ioutil.WriteFile(path, index, 0644)
+}
+
+// DownloadBundleFile downloads the tgz file from the given repo
+func (r *FrameworkRepository) DownloadBundleFile(bundle string) error {
+	var fileURL string
+	parsedURL, err := url.Parse(r.Config.URL)
+	if err != nil {
+		return errors.Wrap(err, "parsing config url")
+	}
+	parsedURL.Path = parsedURL.Path + "/" + bundle + ".tgz"
+
+	fileURL = parsedURL.String()
+
+	resp, err := r.Client.Get(fileURL)
+	if err != nil {
+		return errors.Wrap(err, "getting file url")
+	}
+
+	err = untar(vars.RepoPath+"/"+bundle, resp)
+	if err != nil {
+		return errors.Wrapf(err, "failed unpacking %s", bundle)
+	}
+
+	return nil
+}
+
+// GetFrameworkVersion gets the proper Framework version of a given Framework
+func (r *FrameworkRepository) GetFrameworkVersion(name, path string) (string, error) {
+	frameworkVersionPath := path + "/" + name + "-frameworkversion.yaml"
+	frameworkVersionYamlFile, err := os.Open(frameworkVersionPath)
+	if err != nil {
+		return "", errors.Wrap(err, "failed opening frameworkversion file")
+	}
+
+	frameworkVersionByteValue, err := ioutil.ReadAll(frameworkVersionYamlFile)
+	if err != nil {
+		return "", errors.Wrap(err, "failed reading frameworkversion file")
+	}
+
+	var fv v1alpha1.FrameworkVersion
+	err = yaml.Unmarshal(frameworkVersionByteValue, &fv)
+	if err != nil {
+		return "", errors.Wrapf(err, "unmarshalling %s-frameworkversion.yaml content", name)
+	}
+	return fv.Spec.Version, nil
+}
+
+// GetFrameworkVersionDependencies returns a slice of strings that contains the names of all dependency Frameworks
+// from a given repo in the official GitHub repo
+func (r *FrameworkRepository) GetFrameworkVersionDependencies(name, path string) ([]string, error) {
+	frameworkVersionPath := path + "/" + name + "-frameworkversion.yaml"
+	frameworkVersionYamlFile, err := os.Open(frameworkVersionPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed opening frameworkversion file")
+	}
+
+	frameworkVersionByteValue, err := ioutil.ReadAll(frameworkVersionYamlFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed reading frameworkversion file")
+	}
+
+	var fv v1alpha1.FrameworkVersion
+	err = yaml.Unmarshal(frameworkVersionByteValue, &fv)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling %s-frameworkversion.yaml content", name)
+	}
+	var dependencyFrameworks []string
+	if fv.Spec.Dependencies != nil {
+		for _, v := range fv.Spec.Dependencies {
+			dependencyFrameworks = append(dependencyFrameworks, v.Name)
+		}
+	}
+	return dependencyFrameworks, nil
+}
+
+// Untar takes a destination path and a reader; a tar reader loops over the tarfile
```

Comment:
How about the following?

```
// Read a tarball from a reader `r` and extract it to the destination path `dst`.
```

Consider using this function as a base:

https://github.com/golang/build/blob/2759dfe1c117426ad9f2e8ea4cef2a2088dd734e/internal/untar/untar.go#L31-L123

I would otherwise link to the article from which this seems to come: https://medium.com/@skdomino/taring-untaring-files-in-go-6b07cf56bc07